FarihaIS/VanRoomies
# feat(matches): Integrate matches feature

Welcome to VanRoomies! 👋✈️
Fixes:
#49
#65 

## Description of the change:
Integrates the frontend matches page with the backend. Upon clicking on the matches fragment, it should perform a GET request to retrieve all matches. If there are none, a Toast message is displayed. The user can swipe left (to reject the match) and right (to accept the match).

If the user swipes right, this will automatically perform a POST to create a new conversation with that user with the default message "Hello!"

The user's name should show up as expected in the list of chats as well as at the top when a private chat has been clicked upon. 

Furthermore, the block icon in the ChatFragment has been changed to a delete icon to represent the deleting of chats (this will be implemented later in #75 ).

Real-time notifications are still in the works and are not yet implemented. This will be looked into with https://github.com/FarihaIS/VanRoomies/issues/66.

## How to manually test:

1. Start up MongoDB
2. Import the `testdb.preferences.json` and `testdb.users.json` collections (found on Discord under pinned messages)
4. Run npm run dev inside the backend folder
5. Take one of the `_ids` from any of the users and paste it into `MatchesFragment` and `ChatFragment` in the beginning of `onCreateView`. The `thisUserId` field is hardcoded at the moment.
6. Build and run the app
7. Click on the matches button and swipe left/right to ignore/match with a user.
8. If you match with a user, click on the chat button and you should see a new conversation with that user.
9. Try sending messages. The chats should show up in the chat channel and remain after clicking on another fragment and returning to the chat.

Note from before: After sending a chat, going back to the chat fragment, then clicking on the chat, the recent texts may not show up. You will have to click on another fragment then click back on the chat fragment and then the individual chat. This is because of https://github.com/FarihaIS/VanRoomies/issues/62, which needs to be explored further.

Another note: The matches fragment currently starts the deck from the last not-matched-with user (if any). In the future, we will need to save the state of the matches fragment so that when we return to it, we can pick up from where we leave off.
